### PR TITLE
Add synthtool scripts

### LINKED ISSUE #3711: Setup synth scripts for generating clients
This will enable us to automate the regeneration process outside of the release process.

### PR DESCRIPTION
**Pull request by chingor13**
This will allow us to enable autosynth to get nightly PRs for proto changes.

For most APIs, artman is configured to generate 3 folders (grpc-google-cloud-x, proto-google-cloud-x, and google-cloud-x.

Fixes #3711

### CODE REVIEW

**Comment by chingor13** [comment]
@theacodes These synth scripts are pretty straight forward except for the compute one which uses the artman docker image, but the configs are located in discovery-artifact-manager. We can consider adding an option for it in synthtool, but AFAIK, java/compute is the only API that needs it.

File: google-cloud-clients/google-cloud-asset/synth.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""This script is used to synthesize generated parts of this library."""
+
+import synthtool as s
+import synthtool.gcp as gcp
+import logging
+
+logging.basicConfig(level=logging.DEBUG)
```
Comment (theacodes) [inline google-cloud-clients/google-cloud-asset/synth.py]: You can remove these logging statements.

File: google-cloud-clients/google-cloud-compute/synth.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+from synthtool.sources import git
+
+logging.basicConfig(level=logging.DEBUG)
+
+discovery_url = "https://github.com/googleapis/discovery-artifact-manager.git"
+
+log.debug(f"Cloning {discovery_url}.")
+discovery = git.clone(discovery_url, depth=1)
+
+def generate(config, root_dir):
```
Comment (theacodes) [inline google-cloud-clients/google-cloud-compute/synth.py]: Let's just go ahead and get this upstreamed.

Comment (chingor13) [reply google-cloud-clients/google-cloud-compute/synth.py]: Ok, I'll remove from here and get it into synthtool

**Pull request merged**